tests/test_dojo.py

```diff
@@ -0,0 +1,289 @@
+		expected_output = heading_1 + heading_2 + Office_occupants + heading_3
+		self.assertEqual(output, expected_output)
+
+	def  test_print_allocations_with_output_file(self):
+		self.dojo.create_room("Chetambe", "livingspace")
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.add_person("staff", "george", "wanjala", "Y")
+		self.dojo.print_allocations("test_allocations.txt")
+		self.assertTrue(os.path.isfile("test_allocations.txt"))
+		with open("test_allocations.txt") as outputfile:
+			lines = [line.rstrip('\n') for line in outputfile]
+			self.assertTrue("ROOM NAME: NAIROBI\tROLE\t\tACCOMODATION" in lines)
+			self.assertTrue("George Wanjala \t\tSTAFF\t\t" in lines)
+		os.remove("test_allocations.txt")
+
+	def test_print_unallocated_on_screen(self):
+		#setup mock data
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.add_person("fellow", "george", "wanjala", "Y")
+		#string_with_print_content will hold the output being printed
+		stored_standard_output = sys.stdout
+		string_with_print_content = StringIO()
+		sys.stdout = string_with_print_content
+		#call the function which will print to string_with_print_content...
+		#...instead of printing on the screen
+		self.dojo.print_unallocated()
+		#end the function and revert to the original state of sys.stdout
+		sys.stdout = stored_standard_output
+		output = string_with_print_content.getvalue()
+		#escaped characters such as x1b[33m refer to the color
+		heading_1 = "\x1b[34m\nUNALLOCATED - OFFICES\x1b[0m\n"
+		heading_2 = "\x1b[33m\nPERSON NAME\t\tROLE\n\x1b[0m\n\n"
+		heading_3 = "\x1b[34m\nUNALLOCATED - LIVING SPACES\x1b[0m\n"
+		heading_4 = "\x1b[33m\nPERSON NAME\t\tROLE\n\x1b[0m\n"
+		unallocated_persons = "George Wanjala\t\tFELLOW\n"
+		expected_output = heading_1 + heading_2 + heading_3 + heading_4 + unallocated_persons
+		self.assertEqual(output, expected_output)
+
+	def  test_print_uanllocated_with_output_file(self):
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.add_person("fellow", "george", "wanjala", "Y")
+		self.dojo.print_unallocated("test_unallocated.txt")
+		self.assertTrue(os.path.isfile("test_unallocated.txt"))
+		with open("test_unallocated.txt", "r") as outputfile:
+			lines = [line.rstrip() for line in outputfile]
+			self.assertTrue("PERSON NAME\t\tROLE" in lines)
+			self.assertTrue("George Wanjala\t\tFELLOW" in lines)
+		os.remove("test_unallocated.txt")
+
+class TestRellocatePerson(unittest.TestCase):
+	def setUp(self):
+		self.dojo = Dojo()
+
+	def test_rejects_inavalid_id(self):
+		rellocate_a = self.dojo.rellocate_person("a", "Nairobi")
+		self.assertEqual(rellocate_a, "\nInvalid Person ID\n")
+
+	def test_rejects_non_existent_id(self):
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.add_person("fellow", "bob", "james")
+		self.dojo.create_room("Red", "Office")
+		rellocate_bob = self.dojo.rellocate_person(3, "Red")
+		self.assertEqual(rellocate_bob, "\nInvalid Person ID\n")
```

Comment:
Since the ID is a valid type,and you're testing if it exists, a better msg would be "ID does not exist"